server/app/services/rpc_server.rb

```diff
@@ -25,8 +28,38 @@ def initialize(code, message, backtrace = nil)
 
   attr_reader :handlers
 
-  def initialize
+  def initialize(queue)
+    @queue = queue
     @handlers = {}
+    @counter = 0
+    @processing = false
+    every(REPORT_EVERY) {
+      report_queue
+    }
+  end
+
+  def report_queue
+    if @queue.size > QUEUE_WARN_LIMIT
+      warn "#{@queue.size} messages in queue"
+      info "#{@counter / REPORT_EVERY} requests per second"
+    end
+    @counter = 0
+  end
+
+  def process!
+    @processing = true
+    defer {
```

Comment:
Hmm the `defer` feels a little scary, all of the `@handlers` state is now mutated by the defer thread, not the actor thread. Maybe just keep this in the main actor thread, and perhaps move the queue size reporting stuff to the `WebsocketBackend` instead?